### PR TITLE
chore: add generative_uri

### LINKED ISSUE #209: Add `generative_uri ` to the hyperdata
it was contradiction the link you refer: https://www.pob.studio/learn/metadata-cheat-sheet
opensea is using snake_case, which means `generative_uri` instead of `generativeUri`. it means, doesn't complies with current hyperdata yet. im good with any name, just stick with one style. which one do we use?

it doesn't matter how it likes on hyperdata, since it's our own naming. just add our own naming on hyperdata. done.

_Originally posted by @preschian in https://github.com/kodadot/private-workers/pull/188#discussion_r1658426019_

### PR DESCRIPTION
**Pull request by preschian**
## Context

Add `generative_uri` compability to hyperdata

## Ref

- [x] Closes #209

## PR Type

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring
- [x] Chore

### CODE REVIEW

File: hyperdata/package.json [hunk 1]
```diff
@@ -26,7 +26,7 @@
     "lint:fix": "eslint --cache --ext .ts,.js,.mjs,.cjs . --fix && prettier -c src test -w",
     "prepack": "pnpm run build",
     "release": "pnpm test && changelogen --release && npm publish && git push --follow-tags",
-    "test": "vitest run"
```
Comment (vikiival) [inline hyperdata/package.json:29]: this will leave vitest in watch mode - not preffered as it cannot run in CI

File: hyperdata/src/normalize.ts [hunk 2]
```diff
@@ -71,14 +71,17 @@ export function contentFrom(meta: any, eager?: boolean): Content {
...
   if (!uri) {
     return undefined
   }
 
   const hash = meta.previewHash || meta.iterationHash
-  const previewParam = meta.previewParam || 'fxhash'
+  const previewParam =
+    meta.previewParam ||
+    (meta.external_url?.includes('koda') && 'koda') ||
+    'fxhash'
```
Comment (vikiival) [inline hyperdata/src/normalize.ts:84]: Preview param of this 

```
ipfs://bafybeihxrnjmqs7po3jw3p37csyscvgxuwg2khow5zm46liutxq466q45m/?hash=0x044852b2a670ade5407e78fb2863c51de9fcb96542a07186fe3aeda6bb8a116
```

is `hash`

```suggestion
  const previewParam =
    meta.previewParam || 'hash'
```

File: hyperdata/tests/examples/koda-art-generative.json
Comment (vikiival) [inline hyperdata/tests/examples/koda-art-generative.json:1]: Should this represent `collection` or `NFT`

**Review by vikiival — CHANGES_REQUESTED** [review]

File: hyperdata/src/normalize.ts (hunk 2 above)
Comment (preschian) [reply hyperdata/src/normalize.ts]: updated

File: hyperdata/tests/examples/koda-art-generative.json
Comment (preschian) [reply hyperdata/tests/examples/koda-art-generative.json:1]: updated

File: hyperdata/package.json (hunk 1 above)
Comment (preschian) [reply hyperdata/package.json:29]: vitest able to detect which one development mode or CI
- https://vitest.dev/guide/features.html#watch-mode
- https://github.com/kodadot/workers/blob/21fb3d7f7e74ff00fe50c1fb1022f08c5d24fba9/.github/workflows/test.yml#L22
- https://github.com/kodadot/workers/blob/21fb3d7f7e74ff00fe50c1fb1022f08c5d24fba9/services/image/package.json#L9

File: hyperdata/src/normalize.ts [hunk 3]
```diff
@@ -71,14 +71,17 @@ export function contentFrom(meta: any, eager?: boolean): Content {
...
   if (!uri) {
     return undefined
   }
 
   const hash = meta.previewHash || meta.iterationHash
-  const previewParam = meta.previewParam || 'fxhash'
+  const previewParam =
+    meta.previewParam ||
+    (meta.external_url?.includes('koda') && 'hash') ||
+    'fxhash'
```
Comment (vikiival) [inline hyperdata/src/normalize.ts:84]:
```suggestion
  const previewParam = meta.previewParam || 'hash'
```

**Review by vikiival — APPROVED** [review]
Otherwise LGTM

**Pull request merged**